utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -545,6 +545,42 @@ Status PessimisticTransactionDB::Put(const WriteOptions& options,
   return s;
 }
 
+Status PessimisticTransactionDB::PutEntity(const WriteOptions& options,
+                                           ColumnFamilyHandle* column_family,
+                                           const Slice& key,
+                                           const WideColumns& columns) {
+  {
+    const Status s = FailIfCfEnablesTs(this, column_family);
+    if (!s.ok()) {
+      return s;
+    }
+  }
+
+  {
+    std::unique_ptr<Transaction> txn(BeginInternalTransaction(options));
```

Comment:
Maybe as something nice to do as a tech debt clean up, we can use unique_ptr in other places (Put, Delete, etc.) to avoid `delete txn;` like you did here :)

Also I'm wondering what's the benefit of having local scopes here.

Reply:
You mean the local scopes for the `Status` variables? There are cases where reusing a `Status` object can mask failures in a way that goes undetected by the `ASSERT_STATUS_CHECKED` framework, so it's best not to reuse `Status`es. Some people opt for naming the variables `s1`, `s2`... but I prefer to stick with `s` and use local scopes.